### PR TITLE
*: Unify behavior of both local store and HBase

### PR DESCRIPTION
**Pull request by ngaut**
Currently the behavior of local store and HBase are different. After the PR, both of them would be unified.
keys will be locked automatically if transaction change(maybe) the value of the key.

### CODE REVIEW

**Comment by siddontang** [comment]
You can remove all LockKeys in structure too, I think.

**Comment by ngaut** [comment]
keys will be locked automatically if transaction change(maybe) the value of the key.

**Comment by ngaut** [comment]
PTAL @siddontang @c4pt0r

File: store/localstore/txn.go [hunk 1]
```diff
@@ -38,8 +38,8 @@ type dbTxn struct {
 	store        *dbStore // for commit
 	tid          uint64
 	valid        bool
-	version      kv.Version        // commit version
-	snapshotVals map[string][]byte // origin version in snapshot
+	version      kv.Version          // commit version
+	snapshotVals map[string]struct{} // origin version in snapshot
 }
 
 func (txn *dbTxn) markOrigin(k []byte) error {
```
Comment (c4pt0r) [inline store/localstore/txn.go]: return value is useless.

**Comment by siddontang** [comment]
LGTM

**Comment by c4pt0r** [comment]
LGTM

**Pull request merged**